src/matching/Generic_vs_generic.ml

```diff
@@ -1165,7 +1165,46 @@ and m_for_or_if_comp a b =
   | G.CompIf _, _ ->
       fail ()
 
+(* This just does some special coercions for Terraform.
+   If not a Terraform language, skip straight to `m_literal_inner`.
+   https://developer.hashicorp.com/terraform/language/expressions/type-constraints
+*)
 and m_literal a b =
+  Trace_matching.(if on then print_literal_pair a b);
+  fun tin ->
```

Comment:
You can use with_lang, so you don't have to pass around tin

Reply:
thanks!